epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyModelGroup.java

```diff
@@ -59,9 +59,9 @@
 @SuppressWarnings("rawtypes")
 public class EpoxyModelGroup extends EpoxyModelWithHolder<ModelGroupHolder> {
 
-  protected final List<? extends EpoxyModel<?>> models;
-  /** By default we save view state if any of the models need to save state. */
-  private final boolean shouldSaveViewState;
+  protected final List<EpoxyModel<?>> models;
+
+  private boolean shouldSaveViewState = false;
```

Comment:
before the DSL, the recommended approach to customize this was to subclass the EpoxyModelGroup and override `shouldSaveViewState`. Since that isn't possible with a DSL what do you think of just making this not private?

We will want to make sure that the default handling of setting this based on added models does not override a custom setting, so we can do something like
```
private boolean defaultShouldSaveViewState = false;
public Boolean shouldSaveViewState = null;
````

and then the value we use is `shouldSaveViewState ?: defaultShouldSaveViewState` (i wish this file was in kotlin :P )

Reply:
Ok, I missed it, how about this : 

```
group {
  id(...)
  layout(...)
  shouldSaveViewState(true)
  ...
}
```

It should be possible with :

```
  private boolean shouldSaveViewStateDefault = false;

  @Nullable
  public Boolean shouldSaveViewState = null;

  @NonNull
  public EpoxyModelGroup shouldSaveViewState(boolean shouldSaveViewState) {
    onMutation();
    this.shouldSaveViewState = shouldSaveViewState;
    return this;
  }

  @Override
  public boolean shouldSaveViewState() {
    // By default state is saved if any of the models have saved state enabled.
    // Override this if you need custom behavior.
    if (shouldSaveViewState != null) {
      return shouldSaveViewState;
    } else {
      return shouldSaveViewStateDefault;
    }
  }
```

ps: I was really close to migrate the class to Kotlin :)

Reply:
That sounds great